stubs/date.stub

```diff
@@ -9,3 +9,27 @@ class DateInterval
 	public $days;
 
 }
+
+/**
+ * @template TEnd of ?DateTimeInterface
```

Comment:
I guess you need to define a type here without null and add a `?` to the `@return` in the method

Reply:
Or implement `TemplateNullType`.

Reply:
I removed type limitation
@ondrejmirtes `TemplateNullType` ?

Reply:
Remove type limitation is not a solution, we lose the default behavior (https://github.com/phpstan/phpstan-src/pull/958/commits/89d17c745d3c2154415867c8879f9dee97c5cc6a)